pkg/controller/podautoscaler/horizontal.go

```diff
@@ -261,6 +269,9 @@ func (a *HorizontalController) computeReplicasForMetrics(ctx context.Context, hp
 		setCondition(hpa, autoscalingv2.ScalingActive, v1.ConditionFalse, "InvalidSelector", errMsg)
 		return 0, "", nil, time.Time{}, fmt.Errorf(errMsg)
 	}
+	if err := a.validateSelector(hpa, selector); err != nil {
```

Comment:
nit: the previous ifs on `scale.Status.Selector` are also validation, what you add here is yet another validation step. So let's call it e.g. `validateSelectorNotAmbiguous`.

Actually you could hide these 3 selector ifs inside a `validateSelector` support function.

Reply:
Good point, moved all validation into newly introduced function.